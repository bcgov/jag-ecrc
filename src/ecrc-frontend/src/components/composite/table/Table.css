table {
  width: 100%;
}

th {
  background: white;
  padding: 3px;
  border: 5px solid white;
}
table tr:nth-child(even) {
  background: rgba(241, 244, 250, 1);
}

table tr:nth-child(odd) {
  background: rgba(225, 232, 242, 1);
}
