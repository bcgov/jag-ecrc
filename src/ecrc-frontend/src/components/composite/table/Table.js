import React from "react";
import PropTypes from "prop-types";

import "./Table.css";
import TableElement from "../../base/tableelement/TableElement";

export default function Table({ table: { header, tableElements } }) {
  const tableComponents = tableElements.map(element => {
    return <TableElement key={element.name} element={element} />;
  });

  return (
    <table>
      <thead>
        <tr>
          <th colSpan="2">{header}</th>
        </tr>
      </thead>
      <tbody>{tableComponents}</tbody>
    </table>
  );
}

Table.propTypes = {
  table: PropTypes.shape({
    header: PropTypes.string.isRequired,
    tableElements: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired
      }).isRequired
    )
  }).isRequired
};
