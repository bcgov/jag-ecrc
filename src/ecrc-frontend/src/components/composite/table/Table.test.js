import React from "react";
import { create } from "react-test-renderer";

import Table from "./Table";

describe("Table Component", () => {
  test("Matches the snapshot", () => {
    const tableElements = [
      { name: "Org name", value: "Test Org Name" },
      {
        name: "Org Address",
        value: `123 Somewhere Lane\nNo where\nBritish Columbia\nV9V 9V9\nCanada`
      },
      { name: "Applicant Relationship", value: "Employee" }
    ];

    const table = {
      header: "Org Info",
      tableElements
    };

    const simpleTable = create(<Table table={table} />);
    expect(simpleTable.toJSON()).toMatchSnapshot();
  });
});
