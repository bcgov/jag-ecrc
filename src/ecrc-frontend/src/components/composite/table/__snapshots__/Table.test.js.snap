// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Matches the snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th
        colSpan="2"
      >
        Org Info
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Org name
      </td>
      <td>
        Test Org Name
      </td>
    </tr>
    <tr>
      <td>
        Org Address
      </td>
      <td>
        123 Somewhere Lane
No where
British Columbia
V9V 9V9
Canada
      </td>
    </tr>
    <tr>
      <td>
        Applicant Relationship
      </td>
      <td>
        Employee
      </td>
    </tr>
  </tbody>
</table>
`;
